### PR TITLE
feat: created the hamburger component

### LINKED ISSUE #8: Implement A Hamburger Menu Component
Description:
Develop a fully functional Hamburger Menu component as per the design provided in the [FIGMA FILE](https://www.figma.com/design/pL9HcD5saTR6TtMsZYyS7z/STARKNET-UI-KITS?node-id=0-1&node-type=canvas&t=gT1lFVCNijoXRGlC-0) along with all it’s variants. The component showcases an all in one pop up for transaction tracker, network switcher, add token etc.

Requirements:

•⁠  ⁠Design: Follow the provided Figma design precisely for both light and dark mode. Ensure that the component is fully responsive across different screen sizes.
•⁠  ⁠Style Guide: Adhere strictly to the project’s existing style guide for consistency in UI and UX.
•⁠  ⁠Functionality: The component should be fully functional, allowing users to seamlessly switch networks.

After building the component, create a sub-route in the preview route for the component and display each of it’s variant for easy testing. 

Guidance:
There is an already implemented Connect Button component which can serve as a reference for styling, functionality and previewing.

Note:
No other external dependencies should be installed other than the ones already provided.

### PR DESCRIPTION
**Pull request by Dprof-in-tech**
closes #8 

- [x] component is fully functional
- [x]  sub route created for previewing variants
- [x]  component is fully responsive 

below is what this component looks like

Screenshot 2024-09-27 at 09 20 21
Screenshot 2024-09-27 at 09 20 32

### CODE REVIEW

File: app/components/hamburger/hamburger.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+          <button
+            className={`flex w-full items-center justify-center gap-2 py-4 md:h-[60px] md:py-0 md:font-[700] ${
+              theme === "dark"
+                ? "rounded-xl border-[1.75px] border-grey-700 bg-grey-600"
+                : "rounded-xl border border-[#9a9a9a]"
+            } `}
+          >
+            {theme === "dark" ? (
+              <Image
```
Comment (stephanniegb) [inline app/components/hamburger/hamburger.tsx:148]: Use icons from lucide-react instead of images (library big and chevrons)

**Review by stephanniegb — CHANGES_REQUESTED** [review]

File: app/components/hamburger/hamburger.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+const Hamburger: React.FC<HamburgerProps> = ({ theme }) => {
+  const [open, setOpen] = useState(false);
+
+  return (
+    <div className="relative">
+      <button
+        className="z-10 flex h-8 w-8 cursor-pointer flex-col justify-around border-none bg-transparent p-0 focus:outline-none"
+        onClick={() => setOpen(!open)}
+      >
+        <div
```
Comment (Oshioke-Salaki) [inline app/components/hamburger/hamburger.tsx:98]: Fix the opened state of the hamburger it doesn't translate well.

Comment (Oshioke-Salaki) [inline app/components/hamburger/hamburger.tsx:98]: Please fix the translate in the hambutger icon. The open state looks a bit bent.

**Review by Oshioke-Salaki — CHANGES_REQUESTED** [review]

File: app/components/hamburger/hamburger.tsx (hunk 2 above)
Comment (Oshioke-Salaki) [reply app/components/hamburger/hamburger.tsx:98]: also revert changes on package.json

File: app/components/hamburger/hamburger.tsx (hunk 1 above)
Comment (Dprof-in-tech) [reply app/components/hamburger/hamburger.tsx:148]: fixed

File: app/layout.tsx [hunk 3]
```diff
@@ -18,6 +19,7 @@ export default function RootLayout({
       <StarknetProvider>
         <body className={`p-8 antialiased`}>
           <Connect layout="grid" theme="light" />
+          <Hamburger theme="dark" />
```
[the remark this answers is not in the document]
Comment (Dprof-in-tech) [reply app/layout.tsx:22]: done

**Comment by Dprof-in-tech** [comment]
i'm working on the transaction history component now. please dont merge yet.

**Comment by Dprof-in-tech** [comment]
Screenshot 2024-09-30 at 20 04 26


i've created the transaction history component

File: app/components/transaction-history/transaction-history.tsx
Comment (Oshioke-Salaki) [inline app/components/transaction-history/transaction-history.tsx:1]: 1. Create a button to handle the opening of the modal. 
2. Make the close button functional.
3. Edit your hamburger component to make use of the transaction history component.

File: public/icons/transaction-history-white.svg
Comment (stephanniegb) [inline public/icons/transaction-history-white.svg:1]: Please delete the images since they are not in use.

**Review by stephanniegb — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (Oshioke-Salaki) [inline package-lock.json:1]: Please revert this change.

File: app/components/hamburger/hamburger.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      <div
+        className={`flex h-[50px] w-full cursor-pointer items-center justify-between rounded-xl px-4 ${
+          theme === "dark"
+            ? "border-[3px] border-grey-700 bg-[#2a2a2a] text-white"
+            : "border-[3px] border-[#eaeaea] bg-[#f0f0f0] text-black"
+        } md:font-[700]`}
+        onClick={() => setIsOpen(!isOpen)}
+      >
+        <span>{selected}</span>
+        <span>{isOpen ? "▲" : "▼"}</span>
```
Comment (stephanniegb) [inline app/components/hamburger/hamburger.tsx:48]: Please use lucide-react chevrons

File: app/components/hamburger/hamburger.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+interface HamburgerProps {
+  theme: "dark" | "light";
+}
+
+const STARKNET_CHAINID = {
+  SN_MAIN: "0x534e5f4d41494e", // Mainnet
+  SN_SEPOLIA: "0x534e5f5345504f4c4941", // Sepolia testnet
+};
+
+const CustomSelect: React.FC<{ theme: "dark" | "light" }> = ({ theme }) => {
```
Comment (stephanniegb) [inline app/components/hamburger/hamburger.tsx:15]: Add dark mode for drop-down

**Review by Oshioke-Salaki — CHANGES_REQUESTED** [review]

**Review by stephanniegb — CHANGES_REQUESTED** [review]

File: public/icons/transaction-history-white.svg
Comment (Dprof-in-tech) [reply public/icons/transaction-history-white.svg:1]: already deleted maam

File: package-lock.json
Comment (Dprof-in-tech) [reply package-lock.json:1]: deleted

File: app/components/hamburger/hamburger.tsx (hunk 5 above)
Comment (Dprof-in-tech) [reply app/components/hamburger/hamburger.tsx:15]: Screenshot 2024-10-01 at 12 18 30


please, there's no dark mode

Comment (stephanniegb) [reply app/components/hamburger/hamburger.tsx:16]: Just use the dark background colors for it, please.

Comment (Dprof-in-tech) [reply app/components/hamburger/hamburger.tsx:16]: Screenshot 2024-10-03 at 13 16 40


done

**Comment by Dprof-in-tech** [comment]
Thank you for your reviews
